contracts/migration/SushiMigrator.sol

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity 0.8.10;
+
+import "../libraries/SafeERC20.sol";
+import "../interfaces/ITreasury.sol";
+import "../interfaces/IUniswapV2Pair.sol";
+import "../interfaces/IUniswapV2Router.sol";
+import "../types/OlympusAccessControlledV2.sol";
+
+/// @notice This contract handles migrating lps from sushiswap to uniswap v2
+///         to make use of this contract ensure contract address is RESERVEMANAGER IN OHM Treasury
+contract SushiMigrator is OlympusAccessControlledV2 {
+    using SafeERC20 for IERC20;
+
+    struct Amounts {
+        uint128 sushiLpBeforeMigration;
+        uint128 leftoverSushiLpAfterMigration;
+        uint128 uniPoolToken0AddedToPool;
+        uint128 uniPoolToken1AddedToPool;
+        uint128 uniPoolLpReceived;
+        uint128 uniPoolToken0ReturnedToTreasury;
+        uint128 uniPoolToken1ReturnedToTreasury;
+    }
+
+    uint256 public txCount;
+
+    mapping(uint256 => Amounts) public amountsByMigrationId;
+
+    address immutable treasury = 0x9A315BdF513367C0377FB36545857d12e85813Ef;
+
+    constructor(address _olympusAuthority) OlympusAccessControlledV2(IOlympusAuthority(_olympusAuthority)) {}
+
+    /// @notice Takes lp from treasury, remove liquidity from sushi, adds liquidity to uniswap v2
+    /// @param sushiRouter_ sushi router addres
+    /// @param uniRouter_ uni router address
+    /// @param sushiLpAddress_ sushi lp address
+    /// @param uniswapLpAddress_ uni lp address
+    /// @param amount_ lp amount to get from treasury and amount of liquidity to remove
+    /// @param removeLiquiditySlippage_ slippage used when removing liquidity....i.e 95% will be 950
+    /// @param addLiquiditySlippage_ slippage used when adding liquidity....i.e 95% will be 950
+    function executeTx(
+        address sushiRouter_,
+        address uniRouter_,
+        address sushiLpAddress_,
+        address uniswapLpAddress_,
+        uint256 amount_,
+        uint256 removeLiquiditySlippage_,
+        uint256 addLiquiditySlippage_
+    ) external onlyGovernor {
+        ITreasury(treasury).manage(sushiLpAddress_, amount_);
+
+        uint256 amount = IUniswapV2Pair(sushiLpAddress_).balanceOf(address(this));
+
+        (uint256 amountOHM, uint256 amountToken) = removeLiquidity(
+            sushiLpAddress_,
+            sushiRouter_,
+            amount,
+            removeLiquiditySlippage_
+        );
+        uint256 amountAfterTx = IUniswapV2Pair(sushiLpAddress_).balanceOf(address(this));
```

Comment:
Shouldn't need this, all of `amount_` should be removed, should not be any left over liquidity